admin_guide/scheduling/scheduler.adoc

```diff
@@ -0,0 +1,827 @@
+endif::openshift-dedicated[]
+and xref:../../admin_guide/managing_projects.adoc#using-node-selectors[node selector]
+set up in your environment.
+//tag::node-selectors[]
++
+For example, make sure that your pod configuration features the `nodeSelector`
+value indicating the desired label:
++
+----
+apiVersion: v1
+kind: Pod
+spec:
+  nodeSelector:
+    <key>: <value>
+...
+----
+
+. Modify the master configuration file
+(*_/etc/origin/master/master-config.yaml_*) in two places:
++
+.. Add `nodeSelectorLabelBlacklist` to the `admissionConfig` section with
+the labels that are assigned to the node hosts you want to deny pod placement:
++
+====
+----
+...
+admissionConfig:
+  pluginConfig:
+    PodNodeConstraints:
+      configuration:
+        apiversion: v1
+        kind: PodNodeConstraintsConfig
+        nodeSelectorLabelBlacklist:
+          - kubernetes.io/hostname
+          - <label>
+...
+----
+====
+
+.. Then, add the same to the `kubernetesMasterConfig` section to restrict direct pod creation:
++
+====
+----
+...
+kubernetesMasterConfig:
+  admissionConfig:
+    pluginConfig:
+      PodNodeConstraints:
+        configuration:
+          apiVersion: v1
+          kind: PodNodeConstraintsConfig
+          nodeSelectorLabelBlacklist:
+            - kubernetes.io/hostname
+            - <label_1>
+...
+----
+====
+
+. Restart {product-title} for the changes to take effect.
+ifdef::openshift-origin[]
++
+----
+# systemctl restart origin-master
+----
+endif::[]
+ifdef::openshift-enterprise[]
++
+----
+# systemctl restart atomic-openshift-master
+----
+endif::[]
+
+== Control Pod Placement to Projects
+
+The _PodNodeSelector_ admission controller allows you to force pods onto nodes associated with a specific project and prevent pods from being scheduled in those nodes. 
+
+The *Pod Node Selector* admission controller determines where a pod can be placed using xref:../../architecture/core_concepts/pods_and_services.adoc#labels[labels on projects] and node selectors specified in pods. After the pod is created, the node selectors are merged into the pod so that the pod specification will include the labels originally included in the specification and any new labels from the node selectors. The example below illustrates the merging effect.  
+
+The *PodNodeSelector* admission controller also allows you to create a list of labels that are permitted in a specific project. This list acts as a _whitelist_ that lets developers know what node selectors are acceptable to use in a project and gives administrators greater control over labelling in a cluster.
+
+To schedule pods onto specific project nodes:
+
+. Activate the *PodNodeSelector* admission controller:
++
+.. Modify the master configuration file (/etc/origin/master/master-config.yaml) to add the admission controller plug-in and labels:
++
+----
+admissionConfig:
+  pluginConfig:
+    PodNodeSelector:
+      configuration:
+        apiversion: v1
+        kind:podNodeSelectorPluginConfig: <1>
```

Comment:
@mburke5678 I realized this configuration is not right for this plugin pod node selector. Right way to configure this is:


```
admissionConfig:
  pluginConfig:
    PodNodeSelector:
      configuration:
        podNodeSelectorPluginConfig:
          clusterDefaultNodeSelector: "k3=v3"
          ns1: region=west,env=test,infra=fedora,os=fedora
```

Or 

```
admissionConfig:
  pluginConfig:
    PodNodeSelector:
      location: <path-to-file>
```

Where the file will look like:

```
podNodeSelectorPluginConfig:
    clusterDefaultNodeSelector: "k3=v3"
     ns1: region=west,env=test,infra=fedora,os=fedora
```